maskrcnn_benchmark/data/datasets/cityscapes.py

```diff
@@ -0,0 +1,236 @@
+                continue
+
+            filtered_boxes.append(box)
+            filtered_segmentations.append(segmentation)
+            filtered_labels.append(label)
+
+        if len(filtered_boxes) < 1:
+            filtered_boxes = torch.empty(0, 4)
+
+        return filtered_boxes, filtered_segmentations, filtered_labels
+
+    def _processPolygons(self, ann):
+        # For a single object polygon annotations are stored in CityScapes like
+        # [[x1, y1], [x2, y2]...] and we need them in the following format:
+        # [x1, y1, x2, y2, x3, y3 ...]
+        polys = []
+        labels = []
+        boxes = []
+
+        def poly_to_tight_box(poly):
+            xmin = int(min(poly[::2]))
+            ymin = int(min(poly[1::2]))
+            xmax = int(max(poly[::2]))
+            ymax = int(max(poly[1::2]))
+            bbox = xmin, ymin, xmax, ymax
+            return bbox
+
+        for inst in ann["objects"]:
+            label = inst["label"]
+            if label not in self.CLASSES:
+                continue
+
+            label = self.name_to_id[label]
+
+            cityscapes_poly = inst["polygon"]
+            poly = []
+            for xy in cityscapes_poly:
+                # Equivalent with `poly += xy` but this is more verbose
+                x = xy[0]
+                y = xy[1]
+                poly.append(x)
+                poly.append(y)
+
+            # In CityScapes instances are described with single polygons only
+            box = poly_to_tight_box(poly)
+
+            boxes.append(box)
+            polys.append([poly])
+            labels.append(label)
+
+        if len(boxes) < 1:
+            boxes = torch.empty(0, 4)
+
+        return boxes, polys, labels
+
+    def _processBinayMasks(self, ann):
+        boxes = []
+        masks = []
+        labels = []
+
+        def mask_to_tight_box(mask):
+            a = mask.nonzero()
+            bbox = [
+                torch.min(a[:, 1]),
+                torch.min(a[:, 0]),
+                torch.max(a[:, 1]),
+                torch.max(a[:, 0]),
+            ]
+            bbox = list(map(int, bbox))
+            return bbox  # xmin, ymin, xmax, ymax
+
+        # Sort for consistent order between instances as the polygon annotation
+        instIds = torch.sort(torch.unique(ann))[0]
+        for instId in instIds:
+            if instId < 1000:  # group labels
+                continue
+
+            mask = ann == instId
+            label = int(instId / 1000)
+            label = self.cityscapesID_to_ind[label]
+            box = mask_to_tight_box(mask)
+
+            boxes.append(box)
+            masks.append(mask)
+            labels.append(label)
+
+        return boxes, masks, labels
+
+    def __len__(self):
+        return len(self.img_paths)
+
+    def get_img_info(self, index):
+        # Reverse engineered from voc.py
+        # All the images have the same size
+        return {
+            "height": 1024,
+            "width": 2048,
```

Comment:
Hi, if a dataset *(in Cityscapes format e.g. [IDD](https://idd.insaan.iiit.ac.in/))* contains some of the images of different resolution, would it affect anything e.g. training?